jena-extras/jena-serviceenhancer/src/main/java/org/apache/jena/sparql/service/enhancer/impl/OpServiceInfo.java

```diff
@@ -93,7 +93,7 @@ public OpServiceInfo(OpService opService) {
 
         // Handling of a null supOp - can that happen?
         Set<Var> visibleSubOpVars = OpVars.visibleVars(baseSubOp);
-        this.visibleSubOpVarsScopedToNorm = VarScopeUtils.normalizeVarScopesGlobal(visibleSubOpVars);
```

Comment:
This one-liner fixes a scoping issue @LorenzBuehmann found just after he reported that this PR works while trying variants of the query that led to this PR. Instead of cut-off results it may lead to missing columns.